static/app/views/onboarding/components/createProjectsFooter.tsx

```diff
@@ -72,7 +73,11 @@ export default function CreateProjectsFooter({
       responses.forEach(p => (nextState.platformToProjectIdMap[p.platform] = p.slug));
       setPersistedOnboardingState(nextState);
 
-      responses.forEach(ProjectActions.createSuccess);
+      responses.forEach(ProjectsStore.onCreateSuccess);
+
+      // Reload org after project creation
+      await fetchOrganizationDetails(api, organization.slug, true, false);
```

Comment:
Let's change `onCreateSuccess` so that it accepts everything it needs to do `fetchOrganizationDetails` itself. I'm thinking code to synchronous state dependencies between stores shouldn't live in random components and should be centralized to those stores.

eg.
```
ProjectsStore.onCreateSuccess(responses, api, organization)
```

Reply:
This is tricky because we need to make sure the request completes, if the api we pass is scoped to a component then it may cancel the request if the component unmounts

I do agree though